src/main/java/org/apache/commons/imaging/ImageFormats.java

```diff
@@ -16,40 +16,68 @@
  */
 package org.apache.commons.imaging;
 
+import org.apache.commons.imaging.formats.bmp.BmpImagingParameters;
+import org.apache.commons.imaging.formats.gif.GifImagingParameters;
+import org.apache.commons.imaging.formats.icns.IcnsImagingParameters;
+import org.apache.commons.imaging.formats.ico.IcoImagingParameters;
+import org.apache.commons.imaging.formats.jpeg.JpegImagingParameters;
+import org.apache.commons.imaging.formats.pcx.PcxImagingParameters;
+import org.apache.commons.imaging.formats.png.PngImagingParameters;
+import org.apache.commons.imaging.formats.pnm.PnmImagingParameters;
+import org.apache.commons.imaging.formats.psd.PsdImagingParameters;
+import org.apache.commons.imaging.formats.rgbe.RgbeImagingParameters;
+import org.apache.commons.imaging.formats.tiff.TiffImagingParameters;
+import org.apache.commons.imaging.formats.wbmp.WbmpImagingParameters;
+import org.apache.commons.imaging.formats.xbm.XbmImagingParameters;
+import org.apache.commons.imaging.formats.xpm.XpmImagingParameters;
+
+import java.util.function.Supplier;
+
 /**
  * Enum of known image formats.
  */
 public enum ImageFormats implements ImageFormat {
-    UNKNOWN,
-    BMP,
-    DCX,
-    GIF,
-    ICNS,
-    ICO,
-    JBIG2,
-    JPEG,
-    PAM,
-    PSD,
-    PBM,
-    PGM,
-    PNM,
-    PPM,
-    PCX,
-    PNG,
-    RGBE,
-    TGA,
-    TIFF,
-    WBMP,
-    XBM,
-    XPM;
+    UNKNOWN(null),
+    BMP(BmpImagingParameters::new, "bmp", "dib"),
+    DCX(PcxImagingParameters::new, "dcx"),
+    GIF(GifImagingParameters::new, "gif"),
+    ICNS(IcnsImagingParameters::new, "icns"),
+    ICO(IcoImagingParameters::new, "ico"),
+    JBIG2(null),
+    JPEG(JpegImagingParameters::new, "jpg", "jpeg"),
+    PAM(PnmImagingParameters::new, "pam"),
+    PSD(PsdImagingParameters::new, "psd"),
+    PBM(PnmImagingParameters::new, "pbm"),
+    PGM(PnmImagingParameters::new, "pgm"),
+    PNM(PnmImagingParameters::new, "pnm"),
+    PPM(PnmImagingParameters::new, "ppm"),
+    PCX(PcxImagingParameters::new, "pcx", "pcc"),
+    PNG(PngImagingParameters::new, "png"),
+    RGBE(RgbeImagingParameters::new, "rgbe"),
+    TGA(null),
+    TIFF(TiffImagingParameters::new, "tif", "tiff"),
+    WBMP(WbmpImagingParameters::new, "wbmp"),
+    XBM(XbmImagingParameters::new, "xbm"),
+    XPM(XpmImagingParameters::new, "xpm");
+
+    private final String[] extensions;
+
+    ImageFormats(Supplier<? extends ImagingParameters> factory, String ...extensions) {
+        this.extensions = extensions;
+    }
 
     @Override
     public String getName() {
         return name();
     }
 
     @Override
-    public String getExtension() {
-        return name();
+    public String[] getExtensions() {
+        return this.extensions.clone();
+    }
+
+    @Override
+    public String getDefaultExtension() {
+        return this.extensions[0];
```

Comment:
`ImageFormats.UNKNOWN.getDefaultExtension()` throws index out of bounds.

Reply:
Ouch, fixing it. Thanks!